arangod/Aql/AqlValue.cpp

```diff
@@ -1633,46 +1346,45 @@ Range const* AqlValue::range() const {
 }
 
 void AqlValue::erase() noexcept {
+  // VPACK_INLINE must have a value of 0, and VPackSlice::None must be equal
+  // to a NUL byte too
+  static_assert(AqlValueType::VPACK_INLINE == 0,
+                "invalid value for VPACK_INLINE");
+
+  // construct a slice of type None
+  // we will simply zero-initialize the two 64 bit words
   _data.words[0] = 0;
   _data.words[1] = 0;
+
   TRI_ASSERT(isEmpty());
 }
 
 size_t AqlValue::memoryUsage() const noexcept {
-  auto const t = type();
+  auto t = type();
   switch (t) {
-    case VPACK_INLINE:
-    case VPACK_INLINE_INT48:
-    case VPACK_INLINE_INT64:
-    case VPACK_INLINE_UINT64:
-    case VPACK_INLINE_DOUBLE:
-    case VPACK_SLICE_POINTER:
-      return 0;
     case VPACK_MANAGED_SLICE:
       return _data.managedSliceMeta.getLength();
+    case VPACK_MANAGED_STRING:
+      return sizeof(std::string) + _data.managedStringMeta.getLength();
```

Comment:
Nit: this will slightly underreport the memory usage, as we are at least missing the one byte for the null terminator. Instead of adding the string's `size()` here, we could also use its `capacity()`, because the string allocation may be larger than what it stores.

Reply:
Fixed